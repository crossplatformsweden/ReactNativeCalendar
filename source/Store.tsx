import { createStore, applyMiddleware } from 'redux';
import { persistStore } from 'redux-persist';
import { Platform } from 'react-native';
import Raven from 'raven-js';
import createRavenMiddleware from 'raven-for-redux';
import {
  composeWithDevTools,
  RemoteReduxDevToolsOptions,
} from 'remote-redux-devtools';

// Allows us to resolve async actions (promises) with Redux. Example: Fetching data from an API.
import promise from 'redux-promise';

// Logs the previous state, action and the next state in the console.
import logger from 'redux-logger';

// Redux thunk letâ€™s us to return a function inside an action instead returning a object.
import thunk from 'redux-thunk';

import RootReducer from './RootReducer';

// Apply middleware
const middleware = applyMiddleware(
  createRavenMiddleware(Raven),
  thunk,
  promise,
  logger
);

// Will allow the Chrome plugin / VS Redux for Code to work
// https://github.com/zalmoxisus/remote-redux-devtools
const devToolsOptions: RemoteReduxDevToolsOptions = {
  name: Platform.OS,
  port: 8081,
};
const composeEnhancers = composeWithDevTools(devToolsOptions);

// Redux Store that hold the whole application state
const Store = createStore(RootReducer, composeEnhancers(middleware));

// Persist state to app storage
export const configurePersistStore = persistStore(Store);

// Export type for use with TypeScript
export type Store = typeof Store;

// Make store available globally
export default Store;
