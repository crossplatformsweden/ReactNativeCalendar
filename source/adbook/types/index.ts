import { Book } from '../actions';
import Moment from 'moment';

export enum ParkingConstants {
    BOOKING_ADDED = 'BOOKING_ADDED',
    BOOKING_CHANGED = 'BOOKING_CHANGED',
    BOOKING_CANCELLED = 'BOOKING_CANCELLED',
}

export type Book = typeof Book;

export enum IVehicleType {
    Van= 'Van',
    Truck= 'Truck',
    Sedan = 'Sedan',
    SUV= 'SUV',
    /**
     * AKA Family Car, SV: Kombi
     *
     * @type {'StationWagon'}
     * @memberof IVehicleType
     */
    StationWagon= 'StationWagon',
}

interface IVehicle {
    vehicleType: IVehicleType;
    vehicleModel?: string;
    regNumber: string;
}

interface IPerson {
    firstName: string;
    lastName: string;
}

interface ITimePeriod {
    timeStart: Moment.Moment;
    timeEnd: Moment.Moment;
}
interface ILocation {
    lat: number;
    lon: number;
}

export interface IBooking {
    timeperiod: ITimePeriod;
    location: ILocation;
    vehicle: IVehicle;
    adress: string;
    lessee: IPerson;
}

export interface IBookingState {
    booking: IBooking;
}

export interface IParkingAction extends IBookingState {
    type: ParkingConstants;
}

export class Booking implements IBooking {
    constructor (readonly timeperiod: ITimePeriod, readonly location: ILocation,
        readonly vehicle: IVehicle, readonly adress: string, readonly lessee: IPerson ) {

        }
}
