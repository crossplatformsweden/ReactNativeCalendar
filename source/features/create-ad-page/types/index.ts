// import { CreateAd } from '../actions';
// export type CreateAd = typeof CreateAd;
import moment from 'moment';

export enum CreateConstants {
    CREATE_SUCCESS = 'CREATE DID SUCCESS',
    CREATE_UPDATED = 'CREATE DID UPDATE',
    CREATE_CANCEL = 'CREATE DID CANCEL',
}

export interface IUser {
    accessToken: string;
    name: string;
    type: string;
  }

export interface ICreateAdPage {
    fromDate: moment.Moment;
    toDate: moment.Moment;
}

export interface ICreateState {
    createAd: ICreateAdPage;
}

export interface ICreateAction extends ICreateState {
    type: CreateConstants;
}