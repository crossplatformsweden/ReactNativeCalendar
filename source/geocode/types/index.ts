// geocode/types/index
/**
 * Geolocation action constant string keys
 * @namespace
 * @public
 * @typedef GeocodingConstants
 * @memberof Constants
 * @type {Object}
 */
export enum GeocodingConstants {
  GEOCODING_BUSY = 'LÃ¤ser kartboken',
  GEOCODING_SUCCESS = 'GEOCODE = SUCCESS',
  GEOCODING_FAILED = 'GEOCODE = FAILED',
}

/**
 * Response from Google API
 * @typedef IGeocodeReducer
 * @type {Object}
 * @memberof TypesModule
 * @public
 * @export
 * @property {Array<IAddressComponent>?} results
 */
export interface IGeocodeState {
  results?: Array<Result>;
}

export interface IGeocodeAction extends IGeocodeState {
  type: GeocodingConstants;
}

// To parse this data:
//
//   import { Convert, Geocode } from "./file";
//
//   const geocode = Convert.toGeocode(json);
//
// These functions will throw an error if the JSON doesn't
// match the expected interface, even if the JSON is valid.

export interface Geocode {
  results?: Result[];
  status?:  string;
}

export interface Result {
  address_components?: AddressComponent[];
  formatted_address?:  string;
  geometry?:           Geometry;
  place_id?:           string;
  types?:              string[];
}

export interface AddressComponent {
  long_name?:  string;
  short_name?: string;
  types?:      string[];
}

export interface Geometry {
  bounds?:        Bounds;
  location?:      Northeast;
  location_type?: string;
  viewport?:      Bounds;
}

export interface Bounds {
  northeast?: Northeast;
  southwest?: Northeast;
}

export interface Northeast {
  lat?: number;
  lng?: number;
}

// Converts JSON strings to/from your types
// and asserts the results of JSON.parse at runtime
export namespace Convert {
  export function toGeocode(json: string): Geocode {
      return cast(JSON.parse(json), o('Geocode'));
  }

  export function geocodeToJson(value: Geocode): string {
      return JSON.stringify(value, null, 2);
  }

  function cast<T>(obj: any, typ: any): T {
      if (!isValid(typ, obj)) {
          throw `Invalid value`;
      }
      return obj;
  }

  function isValid(typ: any, val: any): boolean {
      if (typ === undefined) return true;
      if (typ === null) return val === null || val === undefined;
      return typ.isUnion  ? isValidUnion(typ.typs, val)
              : typ.isArray  ? isValidArray(typ.typ, val)
              : typ.isMap    ? isValidMap(typ.typ, val)
              : typ.isEnum   ? isValidEnum(typ.name, val)
              : typ.isObject ? isValidObject(typ.cls, val)
              :                isValidPrimitive(typ, val);
  }

  function isValidPrimitive(typ: string, val: any) {
      return typeof typ === typeof val;
  }

  function isValidUnion(typs: any[], val: any): boolean {
      // val must validate against one typ in typs
      return typs.find(typ => isValid(typ, val)) !== undefined;
  }

  function isValidEnum(enumName: string, val: any): boolean {
      const cases = typeMap[enumName];
      return cases.indexOf(val) !== -1;
  }

  function isValidArray(typ: any, val: any): boolean {
      // val must be an array with no invalid elements
      return Array.isArray(val) && val.every(element => {
          return isValid(typ, element);
      });
  }

  function isValidMap(typ: any, val: any): boolean {
      if (val === null || typeof val !== 'object' || Array.isArray(val)) return false;
      // all values in the map must be typ
      return Object.keys(val).every(prop => {
          if (!Object.prototype.hasOwnProperty.call(val, prop)) return true;
          return isValid(typ, val[prop]);
      });
  }

  function isValidObject(className: string, val: any): boolean {
      if (val === null || typeof val !== 'object' || Array.isArray(val)) return false;
      let typeRep = typeMap[className];
      return Object.keys(typeRep).every(prop => {
          if (!Object.prototype.hasOwnProperty.call(typeRep, prop)) return true;
          return isValid(typeRep[prop], val[prop]);
      });
  }

  function a(typ: any) {
      return { typ, isArray: true };
  }

  // @ts-ignore
  function e(name: string) {
      return { name, isEnum: true };
  }

  function u(...typs: any[]) {
      return { typs, isUnion: true };
  }

  // @ts-ignore
  function m(typ: any) {
      return { typ, isMap: true };
  }

  function o(className: string) {
      return { cls: className, isObject: true };
  }

  const typeMap: any = {
      'Geocode': {
          results: u(null, a(o('Result'))),
          status: u(null, ''),
      },
      'Result': {
          address_components: u(null, a(o('AddressComponent'))),
          formatted_address: u(null, ''),
          geometry: u(null, o('Geometry')),
          place_id: u(null, ''),
          types: u(null, a('')),
      },
      'AddressComponent': {
          long_name: u(null, ''),
          short_name: u(null, ''),
          types: u(null, a('')),
      },
      'Geometry': {
          bounds: u(null, o('Bounds')),
          location: u(null, o('Northeast')),
          location_type: u(null, ''),
          viewport: u(null, o('Bounds')),
      },
      'Bounds': {
          northeast: u(null, o('Northeast')),
          southwest: u(null, o('Northeast')),
      },
      'Northeast': {
          lat: u(null, 3.14),
          lng: u(null, 3.14),
      },
  };
}
