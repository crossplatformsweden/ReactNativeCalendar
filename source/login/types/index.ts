// login/types/index
import { FacebookLogin, GoogleLogin, BankIdLogin } from '../actions';

export type FacebookLogin = typeof FacebookLogin;

export type GoogleLogin = typeof GoogleLogin;

export type BankIdLogin = typeof BankIdLogin;

/**
 * A user object created when logging in
 * @typedef IUser
 * @type {Object}
 * @public
 * @export
 * @property {string} accessToken - social provider access token
 * @property {string} name - full name of the user
 * @property {string} picture - profile picture url if any
 * @property {string} type - Facebook, Google etc
 * @public
 */
export interface IUser {
  accessToken: string;
  name: string;
  picture?: string;
  type: string;
  username?: string;
  password?: string;
  email?: string;
  autologin?: boolean;
}

export class User implements IUser {
  constructor(
    readonly accessToken: string,
    readonly name: string,
    readonly picture: string = '',
    readonly type: string,
    readonly username: string = null,
    readonly password: string = null,
    readonly email: string = null,
    readonly autologin: boolean = false
  ) {}
}

/**
 * LoginReducer constant string keys
 *
 * @namespace
 * @public
 * @typedef LoginConstants
 * @type {Object}
 */
export enum LoginConstants {
  /**
   * Login failed or was cancelled
   */
  LOGIN_FAILED = 'LOGIN_FAILED',
  /**
   * Login state: authentication succeeded
   */
  LOGIN_SUCCESS = 'LOGIN_SUCCESS',

  /**
   * Login state: authentication in progress
   */
  LOGIN_BUSY = 'Kollar vem du Ã¤r',

  /**
   * Login state: logged out of app
   */
  LOGGED_OUT = 'LOGGED_OUT',
}

export interface ILoginState {
  /**
   * The user payload recived from login if any
   *
   * @type {IUser}
   * @memberof ILoginReducer
   */
  user?: IUser;
  /**
   * Determines if the user is successfully logged in
   *
   * @type {boolean}
   * @memberof ILoginReducer
   */
  isLoggedIn: boolean;
}

export interface ILoginAction extends ILoginState {
  type: LoginConstants;
}
