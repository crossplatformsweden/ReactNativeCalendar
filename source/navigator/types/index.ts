export interface INavigationConstants {
  /**
   * The login screen
   */
  LOGIN?: Function;
  /**
   * The main screen
   */
  MAIN?: Function;
  /**
   *  The logged out screen
   */
  LOGOUT?: Function;
  /**
   * the map screen
   */
  MAP?: Function;
  /**
   * The profile screen
   */
  PROFILE?: Function;
  /**
   * The booking screen
   */
  BOOK?: Function;
}

export interface INavigationState extends INavigationConstants {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}

/**
 * Navigation constant keys
 */
export enum NavigationConstants {
  /**
   * The login screen
   */
  LOGIN = 'LoginScreen',
  /**
   * The main screen
   */
  MAIN = 'MainScreen',
  /**
   *  The logged out screen
   */
  LOGOUT = 'LogoutScreen',
  /**
   * the map screen
   */
  MAP = 'MapScreen',
  /**
   * The profile screen
   */
  PROFILE = 'ProfileScreen',
  /**
   * The booking screen
   */
  ADBOOK = 'BookingScreen',
}
