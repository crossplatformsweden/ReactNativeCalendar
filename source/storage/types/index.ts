// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
  STORAGE_GOTKEY = 'STORAGE = GOT KEY',
  STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
  STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
  STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
  STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
  STORAGE_SAVED = 'STORAGE = SAVED KEY',
  STORAGE_USER_KEY = 'STORAGE KEY = USER',
  STORAGE_BUSY = 'Friskar upp minnet',
}

export interface IKey {
  /**
   * The key in app storage
   *
   * @type {string}
   */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {
  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
  type: StorageConstants;
}

/**
 * Creates a {IStorageState} object
 *
 * @param type Storage action type const string
 * @param key Storage key
 * @param value Storage value
 */
export const StorageState = (
  type: StorageConstants = null,
  key: string = null,
  value: object = null
): IStorageState => {
  const r: IStorageState = {
    key,
    value,
    type,
  };

  return r;
};
