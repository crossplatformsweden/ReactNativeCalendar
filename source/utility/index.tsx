// utility/index
import UtilityReducer from './reducers';
import * as UtilityTypes from './types';
import SentryUtility from './SentryUtility';
import BusyIndicator, {
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
} from './components/BusyIndicator';
import ComponentBase from './components/ComponentBase';
import { AppErrorChanged, AppLoadingChanged } from './actions';

export {
  AppErrorChanged,
  AppLoadingChanged,
  ComponentBase,
  UtilityReducer,
  UtilityTypes,
  BusyIndicator,
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
  SentryUtility
};
