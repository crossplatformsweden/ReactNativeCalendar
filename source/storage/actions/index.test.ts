// storage/actions/index.test
/// <reference types="jest"/>
import configureMockStore from 'redux-mock-store';

import { Store } from '../../Store';
import * as types from '../types';
import { UtilityTypes } from '../../utility';
import { SaveByKey, GetByKey, RemoveKey } from './';

/**
 * Mock Redux store
 */
const storeMock = configureMockStore<Store>();

// Navigator will call connected components
jest.mock('../../navigator', () => ({
  Routes: 'View',
  AppNavigator: 'Button',
}));

describe('Storage actions', () => {
  beforeEach(() => {
    // @ts-ignore
    storeMock.clearActions();
  });

  test('RemoveKey returns expected actions', async () => {
    const key = 'RemoveKey';
    const methodName = 'RemoveKey';
    const expectedActions = [
      UtilityTypes.AppLoadingChangedAction(
        UtilityTypes.UtilityConstants.APP_LOAD_BUSY,
        types.StorageConstants.STORAGE_BUSY,
        methodName
      ),
      types.StorageState(types.StorageConstants.STORAGE_REMOVEDKEY, key),
      UtilityTypes.AppErrorChangedAction(
        UtilityTypes.UtilityConstants.APP_NO_ERROR,
        null,
        null,
        methodName
      ),
      UtilityTypes.AppLoadingChangedAction(
        UtilityTypes.UtilityConstants.APP_LOAD_DONE,
        null,
        methodName
      ),
    ];

    // @ts-ignore
    await storeMock.dispatch(RemoveKey(key));

    // @ts-ignore
    expect(storeMock.getActions()).toMatchSnapshot();
    // @ts-ignore
    expect(storeMock.getActions()).toEqual(expectedActions);
  });

  test('SaveByKey returns  expected actions', async () => {
    const methodName = 'SaveByKey';
    const model = { name: 'test name', phone: '555-222 33' };
    const testKey = 'test key';

    const expectedActions = [
      UtilityTypes.AppLoadingChangedAction(
        UtilityTypes.UtilityConstants.APP_LOAD_BUSY,
        types.StorageConstants.STORAGE_BUSY,
        methodName
      ),
      types.StorageState(types.StorageConstants.STORAGE_SAVED, testKey, model),
      UtilityTypes.AppErrorChangedAction(
        UtilityTypes.UtilityConstants.APP_NO_ERROR,
        null,
        null,
        methodName
      ),
      UtilityTypes.AppLoadingChangedAction(
        UtilityTypes.UtilityConstants.APP_LOAD_DONE,
        null,
        methodName
      ),
    ];

    // Dispatch action
    // @ts-ignore
    await storeMock.dispatch(SaveByKey(testKey, model));

    // @ts-ignore
    expect(storeMock.getActions()).toMatchSnapshot();
    // @ts-ignore
    expect(storeMock.getActions()).toEqual(expectedActions);
  });

  test('GetByKey returns expected actions', async () => {
    const methodName = 'GetByKey';
    const model = {
      key: methodName,
    };

    const expectedActions = [
      UtilityTypes.AppLoadingChangedAction(
        UtilityTypes.UtilityConstants.APP_LOAD_BUSY,
        types.StorageConstants.STORAGE_BUSY,
        methodName
      ),
      types.StorageState(types.StorageConstants.STORAGE_GOTKEY, methodName, model),
      UtilityTypes.AppErrorChangedAction(
        UtilityTypes.UtilityConstants.APP_NO_ERROR,
        null,
        null,
        methodName
      ),
      UtilityTypes.AppLoadingChangedAction(
        UtilityTypes.UtilityConstants.APP_LOAD_DONE,
        null,
        methodName
      ),
    ];

    // @ts-ignore
    await storeMock.dispatch(GetByKey(methodName));

    // @ts-ignore
    expect(storeMock.getActions()).toMatchSnapshot();
    // @ts-ignore
    expect(storeMock.getActions()).toEqual(expectedActions);
  });
});
