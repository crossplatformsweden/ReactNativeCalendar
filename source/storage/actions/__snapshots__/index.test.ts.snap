// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storage actions GetByKey returns expected actions 1`] = `
Array [
  Object {
    "reason": "Friskar upp minnet",
    "sender": "GetByKey",
    "type": "APP = BUSY LOADING",
  },
  Object {
    "key": "GetByKey",
    "type": "STORAGE = GOT KEY",
    "value": Object {
      "key": "GetByKey",
    },
  },
  Object {
    "exception": null,
    "reason": null,
    "sender": "GetByKey",
    "type": "APP NO ERROR",
  },
  Object {
    "reason": null,
    "sender": "GetByKey",
    "type": "APP = DONE LOADING",
  },
]
`;

exports[`Storage actions RemoveKey returns expected actions 1`] = `
Array [
  Object {
    "reason": "Friskar upp minnet",
    "sender": "RemoveKey",
    "type": "APP = BUSY LOADING",
  },
  Object {
    "key": "RemoveKey",
    "type": "STORAGE = REMOVED KEY",
    "value": null,
  },
  Object {
    "exception": null,
    "reason": null,
    "sender": "RemoveKey",
    "type": "APP NO ERROR",
  },
  Object {
    "reason": null,
    "sender": "RemoveKey",
    "type": "APP = DONE LOADING",
  },
]
`;

exports[`Storage actions SaveByKey returns  expected actions 1`] = `
Array [
  Object {
    "reason": "Friskar upp minnet",
    "sender": "SaveByKey",
    "type": "APP = BUSY LOADING",
  },
  Object {
    "key": "test key",
    "type": "STORAGE = SAVED KEY",
    "value": Object {
      "name": "test name",
      "phone": "555-222 33",
    },
  },
  Object {
    "exception": null,
    "reason": null,
    "sender": "SaveByKey",
    "type": "APP NO ERROR",
  },
  Object {
    "reason": null,
    "sender": "SaveByKey",
    "type": "APP = DONE LOADING",
  },
]
`;
